packages/terra-notification-dialog/src/NotificationDialog.jsx

```diff
@@ -112,179 +103,119 @@ const propTypes = {
 };
 
+const isValidVariant = (variant) => Object.values(variants).indexOf(variant) >= 0;
+
 const actionSection = (acceptAction, rejectAction, buttonOrder, emphasizedAction) => {
-  let acceptButton = null;
-  let rejectButton = null;
   if (!acceptAction && !rejectAction) {
     return null;
   }
+
+  const actionButtons = [];
+
   if (acceptAction) {
-    acceptButton = (emphasizedAction === 'accept')
-      ? <Button text={acceptAction.text} variant={ButtonVariants.EMPHASIS} onClick={acceptAction.onClick} />
-      : <Button text={acceptAction.text} onClick={acceptAction.onClick} />;
-  }
-  if (rejectAction) {
-    rejectButton = (emphasizedAction === 'reject')
-      ? <Button text={rejectAction.text} variant={ButtonVariants.EMPHASIS} onClick={rejectAction.onClick} />
-      : <Button text={rejectAction.text} onClick={rejectAction.onClick} />;
+    const buttonVariant = emphasizedAction === 'accept' ? { variant: ButtonVariants.EMPHASIS } : {};
+    actionButtons.push(<Button {...buttonVariant} data-terra-notification-dialog-button="accept" key="accept" text={acceptAction.text} onClick={acceptAction.onClick} />);
   }
 
-  if (buttonOrder === 'rejectFirst') {
-    return (
-      <div className={cx('actions')}>
-        {rejectButton}
-        {acceptButton}
-      </div>
-    );
+  if (rejectAction) {
+    const buttonVariant = emphasizedAction === 'reject' ? { variant: ButtonVariants.EMPHASIS } : {};
+    actionButtons.push(<Button {...buttonVariant} data-terra-notification-dialog-button="reject" key="reject" text={rejectAction.text} onClick={rejectAction.onClick} />);
   }
 
   return (
     <div className={cx('actions')}>
-      {acceptButton}
-      {rejectButton}
+      {buttonOrder === 'acceptFirst' ? actionButtons : actionButtons.reverse()}
     </div>
   );
 };
 
-const getIcon = (variant, customIcon = null) => {
-  switch (variant) {
-    case variants.ALERT:
-      return (<span className={cx(['icon', 'alert'])} />);
-    case variants.ERROR:
-      return (<span className={cx(['icon', 'error'])} />);
-    case variants.WARNING:
-      return (<span className={cx(['icon', 'warning'])} />);
-    case variants.INFO:
-      return (<span className={cx(['icon', 'info'])} />);
-    case variants.SUCCESS:
-      return (<span className={cx(['icon', 'success'])} />);
-    case variants.CUSTOM:
-      return customIcon;
-    default:
-      return null;
+const NotificationDialog = (props) => {
+  const theme = React.useContext(ThemeContext);
+
+  const {
+    title,
+    startMessage,
+    endMessage,
+    content,
+    acceptAction,
+    rejectAction,
+    variant,
+    isOpen,
+    buttonOrder,
+    emphasizedAction,
+    custom,
+    ...customProps
+  } = props;
+
+  if (isOpen && acceptAction === undefined && rejectAction === undefined) {
+    throw new Error('Unable to render the Notification dialog. Neither an `acceptAction` and/or `rejectAction` prop was provided which leaves the user without the ability to close the dialog.');
```

Comment:
We should reach out to UX for this scenario. Throwing errors for required props should be a last resort.

Does this trigger the application error boundary? If it does the error message will be presented to the user and likely need translations.

Reply:
Initial PR review for reference: https://github.com/cerner/terra-framework/pull/1105#discussion_r422767569

@tbiethman thoughts? I personally don't want to add translations for this. I see Stephen's point for potentially needing them...but I'd really hoping consumers wouldn't ship code with this error. Should we continue to log warning to the console and allow consumers to open the dialog without a way to close it?

Reply:
I agree that adding translations for thrown errors definitely seems like overkill. 

If we know this is going to trigger the error boundary and is displayed face up to the user we might want to evaluate our options/strategy regarding thrown errors that trigger the error boundary. I haven't tested this workflow, but because it's in the render life cycle I do think it will trigger the application error boundary. 

My opinion/vote is not to throw the error at all and let the prop-type checking help developers. Logging a warning should be fine/safe here too if we choose that route.

Reply:
Do you propose we fallback to empty string for the signalWord if the variant is undefined or is not valid?

![Screen Shot 2020-05-13 at 2 04 42 PM](https://user-images.githubusercontent.com/14099737/81854009-e0bb0100-9522-11ea-8346-194b41d4325c.png)

Reply:
The other alternative would be to relax the variant and drop the `isRequired` with the default variant being `custom`. I had originally moved it because @neilpfeiffer  had wanted to make this stricter.

Reply:
I think the screenshot above is what I would expect to happen. The variant is required so prop-type checking should help guide developers. It doesn't code in safety nets to mask the problem and it doesn't crash the app/trigger the error boundary. For sure worth a larger discussion with UX and the team to talk through the advantages and disadvantages of each approach and make a final decision

Reply:
yeah I guess to me, throwing an error that might not be translated seems "safer" than showing a modal with no fallback but both seem bad and should never get to prod. Both are there for developer more or less

Reply:
Having an empty sting fallback leaves an easy out for anyone not implementing the API correctly and leaves an open door for misuse. If variant is missing then the component is broken, and the above visual would be fine to use as the broken state.

Reply:
remove errors: https://github.com/cerner/terra-framework/pull/1105/commits/3ee7c3e4fcb737ea8b167a829b5c9b2ea7d1935e